packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,328 @@
+                    `editing and saving temporary file: ${tempPath}`));
+                }while(input.toLowerCase() !== 'done');{
+                    if (input === null) {
+                        throw new ImperativeError({
+                            msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${tempPath}`)
+                        });
+                    }
+                    return true;
+                }
+            case Prompt.continueToUpload:
+                input = await CliUtils.readPrompt(TextUtils.chalk.green(`Continue uploading edits despite changes on remote? Y/n`));
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${tempPath}`)
+                    });
+                }
+                return input.toLowerCase() === 'y';
+            case Prompt.viewUpdatedRemote:
+                input = await CliUtils.readPrompt(TextUtils.chalk.green(`The remote version of the document you were editing has changed. `+
+                `View the updated version of the mainframe file before proceeding with edits? Y/n`));
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${tempPath}`)
+                    });
+                }
+                return input.toLowerCase() === 'y';
+        }
+    }
+
+    /**
+     * Download file and determine if downloading just to get etag (useStash) or to save file locally & get etag (!useStash)
+     * @param {AbstractSession} session - the session object generated from the connected profile
+     * @param {LocalFile} lfFile - object containing pertinent information about the local file during the editing process
+     * @param {boolean} useStash - should be true if don't want to overwrite local file when refreshing etag
+     * @returns {LocalFile}
+     */
+    public static async localDownload(session: AbstractSession, lfFile: LocalFile, useStash: boolean): Promise<LocalFile>{
+        // account for both useStash|!useStash and uss|ds when downloading
+        const tempPath = useStash ? path.join(tmpdir(), "toDelete.txt") : lfFile.tempPath;
+        const args: [AbstractSession, string, IDownloadOptions] = [
+            session,
+            lfFile.fileName,
+            {
+                returnEtag: true,
+                file: tempPath
+            }
+        ];
+        if(lfFile.fileType === 'uss'){
+            lfFile.zosResp = await Download.ussFile(...args);
+        }
+        lfFile.zosResp = await Download.dataSet(...args);
+        // TODO: get rid of the fake temp file generated by the useStash case (wouldnt have to do this if the option {overwrite: false} worked)
```

Comment:
Good catch on the bug with `overwrite = False` not working - seems like something worth fixing in this PR if its a quick fix, or otherwise creating a separate issue for it 👍

Reply:
after talking this through with Fernando, I've realized the functionality was intended to work in a different way than I thought and expected so it seems there's no issue/bug to be fixed